crates/storage/src/snapshot.rs

```diff
@@ -293,36 +328,41 @@ impl StateRead for Snapshot {
 
     fn nonverifiable_prefix_raw(&self, prefix: &[u8]) -> Self::NonconsensusPrefixRawStream {
         let span = Span::current();
-        let self2 = self.clone();
+        let snapshot = self.clone();
 
+        let (prefix, config) = self.0.multistore.route_key(prefix);
```

Comment:
For the prefix queries, I think we need to do another check: we need to ensure that the iteration doesn't cross a "mount point". I think we can do this by banning queries supplying prefixes that are themselves prefixes of mount points, returning an error instead.  (I don't think this is a meaningful restriction, maybe it means we can't iterate over every single key but I think that's OK)